openedxstats/apps/sites/views.py

```diff
@@ -44,16 +45,53 @@ class OTChartView(JSONResponseMixin, generic.list.MultipleObjectTemplateResponse
     template_name = 'sites/ot_chart.html'
     context_object_name = 'snapshot_list'
 
+    def daterange(self, start_date, end_date):
+        """
+        This function is used to generate a date range, with one day increments. Notice the +1 adjustment on day, and -1
+        adjustment on seconds. This is used to ensure each day is actually a datetime of the last second of that day, to
+        allow for correct aggregation when querying the database with these dates.
+        """
+        for n in range(int((end_date - start_date).days)):
+            yield start_date + timedelta(n+1, seconds=-1)
+
+    def generate_summary_data(self, start_datetime):
+        """
+        Generate site total and course totals by day since ending of site summary snapshots were recorded
+        """
+        daily_summary_obj_list = []
+        for day in self.daterange(start_datetime, datetime.now() + timedelta(days=1)):
+            day_stats = Site.objects.filter(
+                Q(course_count__gt=0) & Q(active_start_date__lte=day) &
+                (Q(active_end_date__gte=day) | Q(active_end_date=None))).values(
+                'active_start_date').aggregate(
+                sites=Count('active_start_date'), courses=Sum('course_count')
+            )
```

Comment:
This at least needs to be formatted so the parens help see the structure, and maybe a comment.